src/okta/createOktaClient.ts

```diff
@@ -0,0 +1,134 @@
+  logger: IntegrationLogger;
+  minimumRateLimitRemaining: number;
+  requestAfter: number | undefined;
+
+  constructor(
+    logger: IntegrationLogger,
+    options?: RequestExecutorWithEarlyRateLimitingOptions,
+  ) {
+    super();
+    this.logger = logger;
+    this.minimumRateLimitRemaining =
+      options?.minimumRateLimitRemaining ||
+      DEFAULT_MINIMUM_RATE_LIMIT_REMAINING;
+  }
+
+  async fetch(request: any) {
+    const now = Date.now();
+    if (this.requestAfter && this.requestAfter > now) {
+      const delayMs = this.requestAfter - now;
+      await new Promise((resolve) => setTimeout(resolve, delayMs));
+    }
+    return super.fetch(request).then((response: any) => {
+      this.requestAfter = this.getRequestAfter(response);
+      return response;
+    });
+  }
+
+  getRateLimitReset(response: any) {
+    return response.headers.get('x-rate-limit-reset');
+  }
+
+  getRateLimitRemaining(response: any) {
+    return response.headers.get('x-rate-limit-remaining');
+  }
+
+  getRequestAfter(response: any) {
+    const rateLimitRemaining = this.getRateLimitRemaining(response);
+    if (rateLimitRemaining <= this.minimumRateLimitRemaining) {
+      const requestAfter =
+        new Date(
+          parseInt(this.getRateLimitReset(response), 10) * 1000,
+        ).getTime() + 1000;
+      this.logger.info(
+        {
+          minimumRateLimitRemaining: this.minimumRateLimitRemaining,
+          requestAfter,
+          url: response.url,
+        },
+        'Minimum `x-rate-limit-remaining` header reached. Temporarily throttling requests',
+      );
+      return requestAfter;
+    }
+    return undefined;
+  }
+}
+
+export default function createOktaClient(
+  logger: IntegrationLogger,
+  config: OktaIntegrationConfig,
+  options?: RequestExecutorWithEarlyRateLimitingOptions,
+): OktaClient {
+  const defaultRequestExecutor = new RequestExecutorWithEarlyRateLimiting(
+    logger,
+    options,
+  );
+
+  defaultRequestExecutor.on(
+    'backoff',
+    (request: any, response: any, requestId: any, delayMs: any) => {
+      logger.info(
+        {
+          delayMs,
+          requestId,
+          url: request.url,
+        },
+        'Okta client backoff',
+      );
+    },
+  );
+
+  defaultRequestExecutor.on('resume', (request: any, requestId: any) => {
+    logger.info(
+      {
+        requestId,
+        url: request.url,
+      },
+      'Okta client resuming',
+    );
+  });
+
+  defaultRequestExecutor.on('request', (request: any) => {
+    logger.trace(
+      {
+        url: request.url,
+      },
+      'Okta client initiated request',
+    );
+  });
+
+  defaultRequestExecutor.on('response', (response: any) => {
+    logger.trace('Okta client received response');
+  });
+
+  return new okta.Client({
+    orgUrl: config.oktaOrgUrl,
+    token: config.oktaApiKey,
+    requestExecutor: defaultRequestExecutor,
+    // is this necessary? We're not re-calling the same APIs.
```

Comment:
I suspect it is not really necessary. It's been in this code since before I came 3 years ago, but I've not made the time to figure out whether it's really being used.